src/dex/m-token/m-token-e2e.test.ts

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-console */
+import dotenv from 'dotenv';
+dotenv.config();
+
+import { testE2E } from '../../../tests/utils-e2e';
+import { Tokens, Holders } from '../../../tests/constants-e2e';
+import { Network, ContractMethod, SwapSide } from '../../constants';
+import { StaticJsonRpcProvider } from '@ethersproject/providers';
+import { generateConfig } from '../../config';
+
+function testForNetwork(
+  network: Network,
+  dexKey: string,
+  tokenASymbol: string,
+  tokenBSymbol: string,
+  tokenAAmount: string,
+  tokenBAmount: string,
+) {
+  const provider = new StaticJsonRpcProvider(
+    generateConfig(network).privateHttpProvider,
+    network,
+  );
+  const tokens = Tokens[network];
+  const holders = Holders[network];
+
+  const sideToContractMethods = new Map([
+    [SwapSide.SELL, [ContractMethod.swapExactAmountIn]],
+    // [SwapSide.BUY, [ContractMethod.swapExactAmountOut]],
+  ]);
+
+  describe(`${network}`, () => {
+    sideToContractMethods.forEach((contractMethods, side) =>
+      describe(`${side}`, () => {
+        const isSell = side === SwapSide.SELL;
+
+        contractMethods.forEach((contractMethod: ContractMethod) => {
+          describe(`${contractMethod}`, () => {
+            it(`${tokenASymbol} -> ${tokenBSymbol}`, async () => {
+              await testE2E(
+                // Changing the source here for AugustV6 to fund the sender account with MToken for SELL (M->WrappedM)
+                // and WrappedM for BUY (WrappedM->M). This is to ensure that the sender account has enough balance for testing
+                isSell ? tokens[tokenASymbol] : tokens[tokenBSymbol],
+                isSell ? tokens[tokenBSymbol] : tokens[tokenASymbol],
+                isSell ? holders[tokenASymbol] : holders[tokenBSymbol],
+                isSell ? tokenAAmount : tokenBAmount,
```

Comment:
what's a better way to tell the E2E utils to fund the accounts differently? 

Breakdown of the issue
- In the case of source=M dest=wM side=BUY
- AugustV6 would fund the sender's account with M tokens (source token)
- And when trying to call unwrap on WrappedM the sender's would consequently have no wM balance

As a result, I had to swap those lines to trick the utilities to fund the sender's account with wM.

Here is the Tenderly log with the issue:

![Screenshot 2025-02-03 at 13 33 48](https://github.com/user-attachments/assets/9fde8391-992c-41da-b36d-07f799cdcfaf)